### ISSUE TITLE
prometheus path annotation not honored for service discovery?

### ISSUE THREAD

**Issue by DoyleRavPMPC** [Azure/prometheus-collector#613]
Hello again,

TL:DR; I've added a metrics path annotation to some of my pods, and I see the annotations picked up by managed prometheus, but the default path is used (and doesn't work).   Am I missing a step in overriding the default path for service discovery with pod annotation?

Details:

I've configured a pod with prometheus.io annotations to enable scraping it.  One of the annotations is path, ie:
`prometheus.io/path: '/api/myservice/telemetry/metrics'
`
I've also set the following:
 ```
        prometheus.io/scrape: 'true'
        prometheus.io/port: '80'
        prometheus.io/scheme: 'http'
```

```
My prometheus config is pretty simple:
global:
  scrape_interval: 15s

scrape_configs:
  - job_name: kubernetes_pods
    kubernetes_sd_configs:
      - role: pod
```

I've set the pod annotation namespace in ama-metrics-settings-configmap to include my service's namespace (as well as nginx-ingress). 

I apply the config for settings and prometheus to the ama-metrics config maps - so far so good.  I wait for a bit for the config to take effect.  My deployment and pod show the expected annotations for prometheus.io.

I use kubectl exec and curl to ensure I see metrics at the given url, ala:
` kubectl exec -n <mynamespace> <mypodthathascurl> curl http://<myserviceip>/api/<myservice>/telemetry/metrics`

I see metrics at the given url.

I then use port-forward to see the prometheus ui on the ama-metrics pod.  I see a lot of targets show up for service discovery, which is good.  I see the nginx metrics show up on 10254, as expected (nginx annotations have scrape=true and port=10254, but does not override the default /metrics path).  I see a bunch of kube-system targets, some of which have annotations apparently and are getting scraped successfully.

I find my pod in targets, but the path listed for the service appears to be the default, not my annotated override:

<html>
<body>

http://10.224.0.173/metrics | DOWN
-- | --

</body>
</html>

I look under service discovery and I see the following relevant annotations for my service:

```
__meta_kubernetes_pod_annotation_prometheus_io_path="/api/myservice/telemetry/metrics"
__meta_kubernetes_pod_annotation_prometheus_io_scrape="true"
__meta_kubernetes_pod_annotationpresent_prometheus_io_path="true"
__meta_kubernetes_pod_annotationpresent_prometheus_io_scrape="true"
...
__metrics_path__="/metrics"
__scheme__="http"
__scrape_interval__="15s"
__scrape_timeout__="10s"
```

Is there something else I need to configure or specify to get my path annotation honored for certain pods that use a non-default metrics path?

Thanks,
Doyle

**Comment by DoyleRavPMPC** [comment]
Update: I needed to add the relabeling here to get the custom metrics path.  Relabeling for prometheus config service discovery job is shown
[here](https://learn.microsoft.com/en-us/azure/azure-monitor/containers/prometheus-metrics-scrape-configuration#pod-annotation-based-scraping)

And now it works!  Thanks to the team and Vishiy for the pointer.
